pyanaconda/payload/dnfpayload.py

```diff
@@ -380,35 +382,39 @@ def _add_repo(self, ksrepo):
         #     kickstarts.
         if repo.id in self._base.repos:
             if not url and not mirrorlist and not metalink:
-                self._base.repos[repo.id].enable()
+                if enabled:
+                    self._base.repos[repo.id].enable()
             else:
                 with self._repos_lock:
                     self._base.repos.pop(repo.id)
                     self._base.repos.add(repo)
-                repo.enable()
         # If the repo's not already known, we've got to add it.
         else:
             with self._repos_lock:
                 self._base.repos.add(repo)
-            repo.enable()
```

Comment:
Why these two `repo.enable()` are not here anymore?

Reply:
I consolidated them into a single repo.enable after the log message below.

Reply:
Pondering a bit more, I shifted the fetch into a second function so I can make a clean `addDisabledRepo` that re-uses most of `_add_repo`